### PR TITLE
IDPF-92 Added redis, updated config files, added support for dbt-platform

### PR DESCRIPTION
**Pull request by 0RWilliams**
Updated Identity service with support for Redis, and  DBT-Platform.
As part of the work to get this service deployed i have added:

- `dbt-copilot-python` for checking the environment and setting up the database connections 
- `dj-database-url` for mapping the database connection string
-  replaced `redis_url` with `IDENTITY_REDIS_URL` and `IDENTITY_REDIS` to match `dbt-terraform` parameters.

I have also created a `prod.py` config file, and updated `manage.py` to default to `prod.py` ready for deployment.

Added Typing to config files (Not sure if we should go down this route, or if it's overkill)
Moved all Env declarations to the top of the file. (discussion point)

*Downgraded waitress to 2.1.2 as we had problems with this upgrade on several other services.

### CODE REVIEW

File: config/settings/base.py [hunk 1]
```diff
@@ -12,30 +12,59 @@
...
 import sentry_sdk
+from dbt_copilot_python.database import database_url_from_env
+from dbt_copilot_python.network import is_copilot, setup_allowed_hosts
 from sentry_sdk.integrations.django import DjangoIntegration
+from sentry_sdk.integrations.redis import RedisIntegration
 
 
 # Build paths inside the project like this: BASE_DIR / 'subdir'.
-BASE_DIR = Path(__file__).resolve().parent.parent.parent
+BASE_DIR: Path = Path(__file__).resolve().parent.parent.parent
```
Comment (marcelkornblum) [inline config/settings/base.py:25]: my own 2p on the ordering thing: personally I'm not a fan. I understand the inclination but IMO there's so much "get this var from the ENV" in the settings that you don't get much benefit from putting it here, and it does mean you are starting to mess up the set of things you need to look at when dealing with all the related settings for something; you'll be scrolling around and it'll make it harder to e.g. see typos etc.

File: config/settings/base.py [hunk 2]
```diff
@@ -74,43 +101,49 @@
...
-            DjangoIntegration(),
-        ],
-        enable_tracing=env.bool("SENTRY_ENABLE_TRACING", False),
-        traces_sample_rate=env.float("SENTRY_TRACES_SAMPLE_RATE", 0.0),
+        dsn=SENTRY_DSN,
+        environment=APP_ENV,
+        release=GIT_COMMIT,
+        integrations=[DjangoIntegration(), RedisIntegration()],
+        enable_tracing=SENTRY_ENABLE_TRACING,
+        traces_sample_rate=SENTRRY_TRACES_SAMPLE_RATE,
```
Comment (marcelkornblum) [inline config/settings/base.py:112]: [continuing previous comment] e.g. here's a typo - but more importantly I don't get to see here what the default value is, I need to scroll up to see it

File: config/settings/base.py [hunk 3]
```diff
@@ -124,26 +157,3 @@ def init_sentry():
         "NAME": "django.contrib.auth.password_validation.NumericPasswordValidator",
     },
 ]
-
```
Comment (marcelkornblum) [inline config/settings/base.py:127]: why did these all go to the top? you're moving all your less-used and less-read stuff above the stuff you will be wanting to see more often. Bear in mind this file will grow over the course of the project.

**Review by marcelkornblum — APPROVED** [review]
Overall it's good.

Personally I'm not a fan of the settings file ordering, and while I approve of typing, I think it's overkill in the settings myself :)

File: config/settings/base.py (hunk 1 above)
Comment (0RWilliams) [reply config/settings/base.py:25]: Yeah i completely understand, I have made some changes around this, just haven't pushed them yet. After reaching out, what the others have said, like you, makes sense. I personally am a fan of declaring first, but have made the changes to move to a 'theme' approach.

File: config/settings/base.py (hunk 2 above)
Comment (0RWilliams) [reply config/settings/base.py:112]: [Same as previous]

File: config/settings/base.py (hunk 3 above)
Comment (0RWilliams) [reply config/settings/base.py:127]: [Same as previous]

**Review by alsyx — APPROVED** [review]

**Pull request merged**